EXTENSIONS.md

```diff
@@ -0,0 +1,236 @@
+
+Function decorator details:
+- First parameter: the function being decorated (or, whatever the next inner decorator returned)
+- Second parameter: a context object which just has `{ kind: 'function' }`
+- Return value: a new function, or undefined to keep the same function
+
+The inner binding of a function expression inside itself is in TDZ until all the function decorators run.
+
+## Parameter decorators and annotations
+
+A parameter decorator wraps the value of a function/method argument. It returns a function which does the wrapping.
+
+```js
+function dec(_, context) {
+  assert(context.kind === "parameter");
+  return arg => arg - 1;
+}
+
+function f(@dec @{x: "y"} arg) { return arg * 2 ; }
+
+f(5)  // 8
+f[Symbol.annotations].parameters[0].x  // "y"
+```
+
+Functions with parameters that are decorated or annotated are treated similarly to decorated/annotated functions: they are not hoisted, and are in TDZ until their definition is executed.
+
+Parameter decorator details:
+- First parameter: undefined
+- Second parameter: a context object which just has `{ kind: 'parameter' }`
+- Return value: a function which takes a parameter value and returns a new parameter value. The function is called with the `this` value that the surrounding function is called with.
+
+This example can be desugared as
+
+```js
+let decInstance = dec(undefined, {kind: "parameter"});
+
+function f(arg_) {
+  const arg = decInstance(arg_);
+  return arg * 2 ;
+}
+
+f[Symbol.annotations] = {}
+f[Symbol.annotations].parameters = []
+f[Symbol.annotations].parameters[0] = {x: "y"};
+```
+
+## `let` decorators
+
+Variables declared with `let` can be decorated, converting them into special getter/setter pairs that are invoked when the variable is read or written.
+
+```js
+let @deprecated x = 1;
+++x;  // Shows deprecation warnings for read and write
+```
+
+`let` decorators might be useful for systems using reactivity based on local variables, e.g. "hooks" systems.
+
+`let` decorator details:
+- First parameter: A `{ get, set, value }` object (where the receiver is expected to be undefined, and where `value` is the RHS)
+- Second parameter: a context object `{ kind: "let" }`
+- Return value: A `{ get, set, value }` object, with potentially new behavior
+
+This example can be desugared as:
+
+```js
+let { get_x, set_x, value: _x } = deprecated({value: 1, get() { return _x; }, set(v) { _x = v; }}, {kind: "let"});
+
+set_x(get_x()++);
+```
+
+## `const` decorators
+
+Variables declared with `const` can be decorated more simply--the decorator simply wraps the value being decorated when it's being defined.
```

Comment:
It should also be able to capture access to allow the previous `@deprecated` to work. But value returned from the `get` should be ignored

Reply:
Yes, that is another possibility within this framework.